lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp

```diff
@@ -51,6 +51,16 @@ namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
+std::vector<std::vector<int64_t>> compute_shape_embedding(const at::Tensor & weight, const at::Tensor & indices, int64_t padding_idx, bool scale_grad_by_freq, bool sparse){
+  // Based on aten/src/ATen/native/Embedding.cpp::embedding.
+  std::vector<int64_t> out_sizes = indices.sizes().vec();
+  out_sizes.emplace_back(weight.size(1));
```

Comment:
I think this is technically different from https://github.com/pytorch/pytorch/blob/a510f4139b2de1f5baed13bd50a035e39fd2a78b/aten/src/ATen/native/Embedding.cpp#L29, which uses `ArrayRef<T>::slice`.

`ArrayRef<T>`::slice looks like it grabs every element starting at weight.size(1) through the end (docs [here](https://github.com/pytorch/pytorch/blob/a510f4139b2de1f5baed13bd50a035e39fd2a78b/c10/util/ArrayRef.h#L207)).

Reply:
Yea, it makes sense.

Reply:
but there is a torch.check that weight is a 2-D tensor,  right?  so it should be equiv?

Reply:
oh yep, good point.